Assets/HoloToolkit/Speech/Scripts/TextToSpeechManager.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+using System;
+using UnityEngine;
+
+#if WINDOWS_UWP
+using Windows.Media.SpeechSynthesis;
+using Windows.Storage.Streams;
+using System.Threading.Tasks;
+#endif
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// Enables text to speech using the Windows 10 <see cref="SpeechSynthesizer"/> class.
+    /// </summary>
+    /// <remarks>
+    /// <see cref="SpeechSynthesizer"/> generates speech as a <see cref="SpeechSynthesisStream"/>. 
+    /// This class converts that stream into a Unity <see cref="AudioClip"/> and plays the clip using 
+    /// the <see cref="AudioSource"/> you supply in the inspector. This allows you to position the voice 
+    /// as desired in 3D space. One recommended approach is to place the AudioSource on an empty 
+    /// GameObject that is a child of Main Camera and position it approximately 0.6 units above the 
+    /// camera. This orientation will sound similar to Cortanas speech in the OS.
+    /// </remarks>
+    public class TextToSpeechManager : Singleton<TextToSpeechManager>
```

Comment:
Though I think it's an outlier scenario, I agree with the concern of not being able to have multiple speech sources playing. Having said that, the way that Singleton is defined [in the toolkit](https://github.com/Microsoft/HoloToolkit-Unity/blob/master/Assets/HoloToolkit/Utilities/Scripts/Singleton.cs) is not like a regular C# singleton. It's actually an instance, and whenever anyone calls .Instance the code uses [FindObjectOfType<T>](http://docs.unity3d.com/ScriptReference/Object.FindObjectOfType.html) to locate the first instance of that object in the Unity scene tree. This means that multiple instances can actually exist within the scene graph. To access each individual one you would just use the "classic" way of calling GetComponent<TextToSpeechManager> on whatever object holds the instance you want to use. This should still work just fine today, but you can also use TextToSpeechManager.Instance as a shortcut to get the first instance in the scene.